modules/core/src/utils/logtagmanager.hpp

```diff
@@ -37,7 +37,21 @@ class LogTagManager
     // also, extensible functions (accepting user-provided callback) are not allowed
     // to call LogTagManger (to prevent iterator invalidation), which needs enforced
     // with a non-recursive mutex.
+#ifndef CV_DISABLE_MULTI_THREAD
     using MutexType = std::mutex;
+#else // CV_DISABLE_MULTI_THREAD
+    // Custom (failing) implementation of `std::mutex`.
+    struct MutexType {
+        void lock(){
```

Comment:
@alalek @vpisarev  - I am seeing failures like the following in my baremetal runs. I have poked around a bit to see if there is a configure option that allows turning off logging, but I haven't found one so far. Should I come up with a "turn off logging" cmake option? Or is there already one and am I simply missing it?

Francesco

```
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.5.2-dev) /home/frapet01/projects/opencv/opencv/modules/core/src/utils/logtagmanager.hpp:46: error: (-213:The function/feature is not implemented) cv::Mutex are disabled by CV_DISABLE_MULTI_THREAD=ON in function 'lock'
```

Reply:
Using `cv::AutoLock` would require to implement stubs for `cv::AutoLock`, which seems to raise issues similar to implementing stub for things like mutexes and threads. 

I think it would make sense to have a separate option that disable logging if invoked at command line configuration. That would be turned on only manually when targeting baremetal. I agree that disabling logging might not be ideal in the general case, but for baremetal there is no filesystem to log to.

Alternatively, if you really think we should use `cv-AutoLock`, I need some help in figuring out what to do. In #19985 the AutoLock/Mutex stubs are defined as follows:

```
#ifndef CV_DISABLE_MULTI_THREAD
typedef std::recursive_mutex Mutex;
typedef std::lock_guard<cv::Mutex> AutoLock;
#else // CV_DISABLE_MULTI_THREAD
// Custom (failing) implementation of `std::recursive_mutex`.
struct Mutex {
    void lock(){
        CV_Error(cv::Error::StsNotImplemented,
                 "cv::Mutex are disabled by CV_DISABLE_MULTI_THREAD=ON");
    }
    void unlock(){
        CV_Error(cv::Error::StsNotImplemented,
                 "cv::Mutex are disabled by CV_DISABLE_MULTI_THREAD=ON");
    }
};
// Stub for cv::AutoLock when threads are disabled.
struct AutoLock {
    AutoLock(Mutex &) { }
};
#endif // CV_DISABLE_MULTI_THREAD
```

What should I based the implementation of AutoLock on? As you can see, even if I use `cv::AutoLock`, I would still be getting exceptions from the stubs of `cv::Mutex`.

Reply:
Right, "no-op" AutoLock stub is required for single-threaded mode ( https://github.com/opencv/opencv/pull/19985#issuecomment-827800993 ).

Reply:
Ah, right! Thanks @alalek . I forgot about the stubs for AutoLock. With this change, it works, no more exceptions :) I have updated the patch for the single thread execution with the suggestion, see #19985 

I have also rebased this one to use that new commit.

One thing that is not clear to me is the reason why we have to raise exceptions for the method of  `cv::Mutex`, and just use stubs for `cv::AutoLock`. I am asking because I would like to add a comment in the code of #19985  explaining this, as other people my ask the same question when seeing the stub and the exception.